store/tikv/2pc.go

```diff
@@ -438,15 +438,18 @@ func (c *twoPhaseCommitter) commitSingleBatch(bo *Backoffer, batch batchKeys) er
 	// Under this circumstance,  we can not declare the commit is complete (may lead to data lost), nor can we throw
 	// an error (may lead to the duplicated key error when upper level restarts the transaction). Currently the best
 	// solution is to populate this error and let upper layer drop the connection to the corresponding mysql client.
-	isPrimary := bytes.Equal(batch.keys[0], c.primary())
+	sender := NewRegionRequestSender(c.store.regionCache, c.store.client)
+	errWrapper := func(err error) error {
+		if err != nil && bytes.Equal(batch.keys[0], c.primary()) && sender.rpcError != nil {
```

Comment:
why not keep `isPrimary := bytes.Equal(batch.keys[0], c.primary())` outside, and use isPrimary inside the closure.

Reply:
If no error occurs, we don't have to check if it contains primary.

Reply:
got